test/testutil.c

```diff
@@ -400,18 +412,17 @@ int test_mem_eq(const char *file, int line, const char *st1, const char *st2,
 
     if (s1 == NULL && s2 == NULL)
         return 1;
-    if (n1 != n2) {
-        test_fail_message(NULL, file, line, "memory",
-                          "size mismatch %s %s [%"OSSLzu"] != %s %s [%"OSSLzu"]",
-                          st1, print_mem_maybe_null(s1, n1, b1), n1,
-                          st2, print_mem_maybe_null(s2, n2, b2), n2);
-        return 0;
-    }
-    if (s1 == NULL || s2 == NULL || memcmp(s1, s2, n1) != 0) {
+    if (n1 != n2 || s1 == NULL || s2 == NULL || memcmp(s1, s2, n1) != 0) {
+        char *m1 = print_mem_maybe_null(s1, n1, b1);
+        char *m2 = print_mem_maybe_null(s2, n2, b2);
+
         test_fail_message(NULL, file, line, "memory",
-                          "%s %s [%"OSSLzu"] != %s %s [%"OSSLzu"]",
-                          st1, print_mem_maybe_null(s1, n1, b1), n1,
-                          st2, print_mem_maybe_null(s2, n2, b2), n2);
+                          "%s %s [%"OSSLzu"] == %s %s [%"OSSLzu"]",
```

Comment:
There seem to be misunderstanding about my last remark with reference to `==` in printout. It was about test_str_eq and test_str_ne, not about *this* function. This function was OK, and now it's wrong. Indeed, it's called to test for equality and error message is emitted when inputs are found to be not equal. Then why does it read `==` now? I mean just try to "listen" to it: "there was error and problem was that 'abcd' was found to be equal to 'efgh'". Really? Hmm... Originally I said I didn't look for more, but now when I look they were *all* like this... Is it really only me who finds them totally misleading?